monkestation/code/modules/blacksmithing/furnace.dm

```diff
@@ -0,0 +1,75 @@
+/obj/structure/furnace
+	name = "furnace"
+	desc = "A furnace."
+	icon = 'monkestation/icons/obj/smith.dmi'
+	icon_state = "furnace0"
+	density = TRUE
+	anchored = TRUE
+	var/debug = FALSE //debugging only
+	var/working = TRUE
+	var/fueluse = 1
+
+
+/obj/structure/furnace/Initialize()
+	..()
+	create_reagents(250, TRANSPARENT)
+	START_PROCESSING(SSobj, src)
+
+/obj/structure/furnace/Destroy()
+	..()
+	STOP_PROCESSING(SSobj, src)
+
+/obj/structure/furnace/process()
+	if(debug)
+		reagents.add_reagent(/datum/reagent/fuel, 1)
+		return TRUE
+	if(reagents.remove_reagent(/datum/reagent/fuel, fueluse))
+		working = TRUE
+		if(icon_state == "furnace0")
+			icon_state = "furnace1"
+	else
+		working = FALSE
+		icon_state = "furnace0"
+
+/obj/structure/furnace/attackby(obj/item/I, mob/user)
```

Comment:
bad var name

Reply:
the obj/item/i or the notsword?

Reply:
replaced notsword hope thats what you meant lol